### PR TITLE
home-manager: fix activate script for use-xdg-base-directories use case

### PR DESCRIPTION
**Pull request by ayum**
### Description

This is illustrative PR for issue https://github.com/nix-community/home-manager/issues/4593

Cannot assure that this is fixing the issue, but for illustrative purpose.

### Checklist

- [x] Change is backwards compatible.

- [ ] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [ ] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

### CODE REVIEW

**Comment by teto** [comment]
thanks for the PR but I think this should use $XDG_ variables instead of $HOME/.local/state etc otherwise it defeats the whole point of xdg

**Comment by ayum** [comment]
> thanks for the PR but I think this should use $XDG_ variables instead of $HOME/.local/state etc otherwise it defeats the whole point of xdg

While i could agree, but corresponding XDG vars may be not defined (and in fact in my configuration they are not defined). May be checks for right command should be even more elaborated.

File: modules/home-environment.nix [hunk 1]
```diff
@@ -585,7 +586,7 @@ in
       if config.submoduleSupport.externalPackageInstall
       then
         ''
-          if [[ -e $HOME/.nix-profile/manifest.json ]] ; then
+          if [[ -e $HOME/.nix-profile/manifest.json ]] || [[ -e $HOME/.local/state/nix/profile/manifest.json ]] ; then
```
Comment (rycee) [inline modules/home-environment.nix:589]: Can use `config.xdg.stateHome`?

```suggestion
          if [[ -e $HOME/.nix-profile/manifest.json || -e "${config.xdg.stateHome}/nix/profile/manifest.json" ]] ; then
```

**Comment by rycee** [comment]
Like @teto said, you need to use the corresponding XDG variable. For example, `${XDG_STATE_HOME:-$HOME/.local/state}`.

**Comment by ayum** [comment]
@teto @rycee 

I have updated PR as you say.

But i can't verify that PR fixes issue. When i do `nix run github:ayum/nix-home-manager/1861d4b3cc2cdd04268bfab5818cf04e936e0214 -- build` (latest PR commit) i cannot see PR changes in `result/activate`. If you guide me how can i test it i will verify PR.

**Comment by ayum** [comment]
@teto @rycee 

I have managed to test this PR and can confirm that it fixes the issue. At least in my configuration with  `use-xdg-base-directories` enabled and no `.nix-profile` in home dir. Home manager profile successfully activates.

Please review changes and approve if there are no more issues with it.

**Comment by Arian04** [comment]
unless I'm misunderstanding the purpose of this PR, wouldn't it be better to just merge #4422?

**Comment by ayum** [comment]
> unless I'm misunderstanding the purpose of this PR, wouldn't it be better to just merge #4422?

@Arian04 
Ah, not seeing it. BTW, #4422 not working for me. I suppose due to line https://github.com/nix-community/home-manager/pull/4594/files#diff-5c980e4c769c045b39cdf1cf9597e9cb2d45929f77e106ee8d9abbcdd7b501a0R210 present in my PR and absent in #4422 .

Aside from this little change #4422 is superior as it touch help message also. If @e-nikolov changes a little that PR i would be totally in favour for it over my PR.

**Comment by ayum** [comment]
@Arian04 @e-nikolov 

I believe that `profileDirectory` not correctly detected in both these PRs.

xdg state home based directory will be used only when we have `nix.enable = true`, nix.settings.use-xdg-base-directories = true`, `xdg.enable = true`, `nix.package = ...` all set.

But it is overly excessive to require that. Nix staff surely have not be required as we may want not manage nix configuration with home manager (my case, i have system wide config, as determinate installer for nix set it for me).

Even xdg stuff should not be required as it is configuration decision. But issue that require fix is about correct detection and profile activation al any possible configuration and it must not depends on user configuration choices.

**Comment by ayum** [comment]
While in current state PR is fixing (narrowly speaking) issue, but wider issue persists. All modules that use `home.profileDirectory` will be somewhat broken unless `.nix-profile` exists in home dir.

More correct detection of profile directory needed.

**Pull request closed without merging**